lib/proxy_api/template.rb

```diff
@@ -11,12 +11,12 @@ def template_url
         return response["templateServer"]
       end
     rescue SocketError, Timeout::Error, Errno::EINVAL, Errno::ECONNRESET,
```

Comment:
Yes, it should have been discussed better, I think the whole feature's terrible.

I don't though wish to accept patches that worsen or entrench what's already a bad piece of code, so yes, I think re-throwing it with a more appropriate message would be best, unless there's a good reason we need to ignore the error.

The only reason I can think of is to try and avoid problems changing a host's build state when the proxy is unavailable, but in that case you're just going to create problems by writing the wrong URL to the templates... and in any case, shouldn't the proxy be available by definition?  Or is this some workaround for its single-threaded nature?